api/docs/release.dox

```diff
@@ -173,6 +177,9 @@ Further non-compatibility-affecting changes include:
  - Added opnd_create_immed_double(), opnd_get_immed_double() and
    opnd_is_immed_double() to enable the creation and handling of double
    precision floating-point operands.
+ - Added -reset_at_thread_count option to allow triggering reset when created thread
+   count reaches the given value. This option is currently used to make Linux reset
+   tests more robust.
```

Comment:
Oh, I misunderstood your previous comment to suggest that we can add an option like that. I should've tried searching for `reset_at_nth_thread` first.

I tried using `reset_at_nth_thread` now. With a value of 2, it gets the assert failure (discussion in other comment thread). With a value higher than that, it doesn't ever reset, because the test has atmost 2 active threads at any point (we do not want the test to have more active threads than that, to keep the output ordering deterministic).

Reply:
I had to keep the new option, because `reset_at_nth_thread` doesn't fail without the stolen reg fix (due to the reason above). But I renamed it to `reset_at_created_thread_count` to highlight the difference from the existing option. Also added a comment with #4496 to remove the new option when we fix the assert failure with `reset_at_nth_thread 2`.

Reply:
Moved these changes to #4498